src/App.tsx

```diff
@@ -2,24 +1,44 @@
+import React, { useState, useCallback } from 'react';
+import { Pagination } from './components/Pagination';
 import './App.css';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const handlePageChange = useCallback((page: number) => {
+    setCurrentPage(page);
+  }, []);
+
+  const handlePerPageChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(e.target.value));
+    setCurrentPage(1); // на 1 ст. при зміні к-ті на ст.
+  };
+
+  const startIndex = (currentPage - 1) * perPage;
+  const paginatedItems = items.slice(startIndex, startIndex + perPage);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startIndex + 1} -{' '}
+        {Math.min(startIndex + perPage, items.length)} of {items.length})
```

Comment:
The expression inside the curly braces can be simplified by using a template string to make the code cleaner.

Reply:
?